File: editor/src/de/project/ice/editor/editors/Editors.java
package de.project.ice.editor.editors;

import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.math.Vector3;
import com.badlogic.gdx.utils.IntMap;
import de.project.ice.utils.Assets;

public abstract class Editors
{
    public static BaseEditor editorForClass(Class type)
    {
        if (Float.class.equals(type) || type == float.class)
        {
            return new NumberEditor.FloatEditor();
        }
        else if (Double.class.equals(type) || type == double.class)
        {
            return new NumberEditor.DoubleEditor();
        }
        else if (Short.class.equals(type) || type == short.class)
        {
            return new NumberEditor.ShortEditor();
        }
        else if (Integer.class.equals(type) || type == int.class)
        {
            return new NumberEditor.IntegerEditor();
        }
        else if (Long.class.equals(type) || type == long.class)
        {
            return new NumberEditor.LongEditor();
        }
        else if (Vector2.class.equals(type))
        {
            return new Vector2Editor();
        }
        else if (Vector3.class.equals(type))
        {
            return new Vector3Editor();
        }
        else if (String.class.equals(type))
        {
            return new StringEditor();
        }
        else if (Assets.Holder.TextureRegion.class.equals(type))
        {
            return new TextureRegionHolderEditor();
        }
        else if (Assets.Holder.Animation.class.equals(type))
        {
            return new AnimationHolderEditor();
        }
        else if (IntMap.class.equals(type))
        {
            return new IntMapEditor();
        }
        else if (OrthographicCamera.class.equals(type))
        {
            return new OrthographicCameraEditor();
        }
        return new BaseEditor();
    }

}

File: editor/src/de/project/ice/editor/editors/Editors.kt
package de.project.ice.editor.editors

import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.math.Vector3;
import com.badlogic.gdx.utils.IntMap;
import de.project.ice.ecs.components.ActivationDirection
import de.project.ice.ecs.systems.CameraSystem
import de.project.ice.screens.CursorScreen
import de.project.ice.utils.Assets;

class Editors {
    companion object {
        fun editorForClass(type: Class<*>): BaseEditor {
            return when(type) {
                Float::class.java, java.lang.Float.TYPE     -> NumberEditor.FloatEditor()
                Double::class.java, java.lang.Double.TYPE   -> NumberEditor.DoubleEditor()
                Short::class.java, java.lang.Short.TYPE     -> NumberEditor.ShortEditor()
                Int::class.java, java.lang.Integer.TYPE     -> NumberEditor.IntegerEditor()
                Long::class.java, java.lang.Long.TYPE       -> NumberEditor.LongEditor()
                Boolean::class.java, java.lang.Boolean.TYPE -> BoolEditor()
                Vector2::class.java                         -> Vector2Editor()
                Vector3::class.java                         -> Vector3Editor()
                String::class.java                          -> StringEditor()
                Assets.Holder.TextureRegion::class.java     -> TextureRegionHolderEditor()
                Assets.Holder.Animation::class.java         -> AnimationHolderEditor()
                IntMap::class.java                          -> IntMapEditor()
                OrthographicCamera::class.java              -> OrthographicCameraEditor()
                CursorScreen.Cursor::class.java             -> CursorEditor()
                CameraSystem.FollowType::class.java         -> FollowTypeEditor()
                ActivationDirection::class.java             -> ActivationDirectionEditor()
                else                                        -> BaseEditor()

            }
        }
    }
}
